File: customers/src/main/java/org/irdigital/cc/customers/api/dtos/CustomerDTO.java
package org.irdigital.cc.customers.api.dtos;

import lombok.Builder;
import lombok.Data;
import lombok.Singular;
import org.irdigital.cc.customers.domain.enums.DocumentType;

import javax.validation.Valid;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;
import java.util.List;

@Data
@Builder
public class CustomerDTO {
    @NotNull
    @NotEmpty
    @Pattern(regexp = "^([a-z]|[A-Z]|'|á|é|í|ó|ú|ñ|ü|Á|É|Í|Ó|Ú|Ñ|Ü|\\s|\\'|\\-)+$")
    private String name;

    @NotNull
    @NotEmpty
    @Pattern(regexp = "^([a-z]|[A-Z]|'|á|é|í|ó|ú|ñ|ü|Á|É|Í|Ó|Ú|Ñ|Ü|\\s|\\'|\\-)+$")
    private String lastName;

    @NotNull
    private DocumentType documentType;

    private String documentTypeDetail;

    @NotNull
    @NotEmpty
    private String identityDocument;

    @NotNull
    @Valid
    @Size(min=1)
    private List<AddressDTO> addresses;

    @NotNull
    @Valid
    @Size(min=1)
    private List<ContactDTO> contacts;

    @Singular
    private List<LinkDTO> links;
}

File: customers-nest/src/customers/dto/customer.dto.ts
import { IsNotEmpty, ArrayNotEmpty, ValidateNested } from "class-validator";
import { Type } from 'class-transformer';
import { ContactDTO } from './contact.dto'
import { AddressDTO } from './address.dto';
import { LinkDTO } from './link.dto';

export class CustomerDTO {
  @IsNotEmpty()
  name: string;
  
  @IsNotEmpty()
  lastName: string;
  
  @IsNotEmpty()
  identityDocument: string;
  
  @IsNotEmpty()
  documentType: string;

  @IsNotEmpty()
  @ArrayNotEmpty()
  @ValidateNested({ each: true})
  @Type(() => ContactDTO)
  contacts: ContactDTO[];

  @IsNotEmpty()
  @ArrayNotEmpty()
  @ValidateNested({ each: true})
  @Type(() => AddressDTO)
  addresses: AddressDTO[];

  links: LinkDTO[] = [];
  addLink(link: LinkDTO) {
    this.links.push(link);
  }
}